csv-generator/csv_generator/consumers/person_consumer.py

```diff
@@ -47,7 +49,7 @@ def process(self, element: 'lxml.etree.ElementTree', xml_file_name: str) -> None
         :param xml_file_name: str
         :return:
         """
-
+        rows = []
```

Comment:
I think the idea is that it can handle large files without keeping everything in memory. It might be better to refactor as a separate ticket if it seems a bottleneck. (my personal preference would be to use generators but there are many ways we could change it).

Reply:
I thought it was just naive code, but it's smelling like premature optimization now.

> I think the idea is that it can handle large files without keeping everything in memory.

opening a file won't read that file into memory. reading the contents of that file, like `open(fname, 'r').read()` *will* read the contents of that file into memory. Opening the file in append-only mode doesn't require reading the contents into memory.

Accumulating rows in memory before opening a file, seeking to the end, and writing it to disk will always be more efficient that writing opening a file, seeking to the end of that file, writing a row, closing that file and then writing another row, *especially* if you're doing everything serially like you are here.

I don't see how generators will improve anything here.

Reply:
Yes, the current approach doesn't keep it in memory but accumulating the rows would, like you said.

A generator could be passed as an iterable to `write_rows` without actually keeping the rows in memory.